src/renderers/webgl/WebGLPrograms.js

```diff
@@ -142,12 +141,12 @@ function WebGLPrograms( renderer, cubemaps, cubeuvmaps, extensions, capabilities
 			vertexShader = material.vertexShader;
 			fragmentShader = material.fragmentShader;
 
-			_customShaders.update( material );
+		}
 
-			customVertexShaderID = _customShaders.getVertexShaderID( material );
-			customFragmentShaderID = _customShaders.getFragmentShaderID( material );
+		_customShaders.update( material, vertexShader, fragmentShader );
```

Comment:
This method call is not without overhead. Is there a noticeable difference for scenes with many render items using built-in materials?

Reply:
I'll update an example to test out.